R/verb-head.R

```diff
@@ -31,21 +31,38 @@ head.tbl_lazy <- function(x, n = 6L, ...) {
   }
   n <- trunc(n)
 
-  if (inherits(x$ops, "op_head")) {
-    x$ops$args$n <- min(x$ops$args$n, n)
+  x$lazy_query <- add_head(x, n)
+  x
+}
+
+add_head <- function(x, n) {
+  lazy_query <- x$lazy_query
+  if (!inherits(lazy_query, "lazy_select_query")) {
+    lazy_query <- lazy_select_query(
+      from = lazy_query,
+      last_op = "head",
+      select = syms(set_names(op_vars(lazy_query))),
```

Comment:
It might be worth using `syms(set_names(op_vars(lazy_query)))` as the default value. I avoided it first to be more explicit and avoid some bugs but this seems to be the standard case.